config.mk

```diff
@@ -24,6 +24,18 @@ CPPFLAGS += -I$(INC)
 # -std=c99 may not be supported by some very old compilers...
 CFLAGS ?= -g -Os -Wall
 
+# Grab the system information
+UNAME_S := $(shell uname -s)
+
+# Set the shared library extension
+ifeq ($(UNAME_S),Darwin)
```

Comment:
This is where you could also set `SOFLAGS` (or whatever you want to call it) that specifies what flags to link shared libs with on the different platforms....